File: SAMSUI/src/app/Model/master/item.ts
import { ItemLocAccessModel } from "../base/itemLocAccess";
import { ModelOfInventory } from "../inventory/Inventory";
import { ModelOfItemMaster } from "../inventory/ItemMaster";
import { ItemApprovedSupplierModel } from "./item-approved-supplier";
export class ItemModel extends ModelOfInventory{

    public itemLocId ?: number;

	public locationId ?: number;

    public invoiceable ?: boolean;

    public itemStatus ?:string;

    public storeId ?: number;

    public maxStockQty ?: number;

    public minStockQty ?: number;

    public costingType ?:string;

    public serialControlled ?: boolean;

    public batchControlled ?: boolean;

	public itemMasterId ?: number;

    public supplierSiteId ?: number;

    public locationName ?:string;

    public itemNameConcat ?:string;

    public itemSupplierList ?: ItemApprovedSupplierModel[];

    public itemLocAccessList ?: ItemLocAccessModel[];
    
	public totalCost ?: number;

	public itemApprovalStatus ?:string;

    public itemMasterTO ?: ModelOfItemMaster;

    public active ?: boolean;

    public activeDisplay ?: string;

    public createdBy ?:string;

    public createdDtDisp ?: string;

    public updatedBy ?: string;

    public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public storeName ?: string;
}

File: sams-backend/src/main/java/com/sams/entity/Item.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_item", schema = "inventory")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "item_loc_id")
    private Long itemLocId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "invoiceable")
    private Boolean invoiceable;

    @Column(name = "item_status")
    private String itemStatus;

    @Column(name = "store_id")
    private Long storeId;

    @Column(name = "max_stock_qty")
    private Integer maxStockQty;

    @Column(name = "min_stock_qty")
    private Integer minStockQty;

    @Column(name = "costing_type")
    private String costingType;

    @Column(name = "serial_controlled")
    private Boolean serialControlled;

    @Column(name = "batch_controlled")
    private Boolean batchControlled;

    @Column(name = "item_master_id")
    private Long itemMasterId;

    @Column(name = "supplier_site_id")
    private Long supplierSiteId;

    @Column(name = "location_name")
    private String locationName;

    @Column(name = "item_name_concat")
    private String itemNameConcat;

    @Column(name = "item_supplier_list")
    private String itemSupplierList;

    @Column(name = "item_loc_access_list")
    private String itemLocAccessList;

    @Column(name = "total_cost")
    private Double totalCost;

    @Column(name = "item_approval_status")
    private String itemApprovalStatus;

    @Column(name = "item_master_t_o")
    private String itemMasterTO;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "active_display")
    private String activeDisplay;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "store_name")
    private String storeName;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}